assets/js/reducers/tictactoeFSM.js

```diff
@@ -1,19 +1,21 @@
 
-import { GameRoundState, GameEvents } from '../actions'
+import { GameRoundState, GameEvents, setGameId, updateGameRound } from '../actions'
 
+const empty = "EMPTY"
 const initialState = {
     gameState: GameRoundState.NOT_STARTED,
-    board: [0, 0, 0, 0, 0, 0, 0, 0, 0]
+    board: [empty, empty, empty, empty, empty, empty, empty, empty, empty]
 }
 
 const allTransitions = [
-    { event: GameEvents.START_ROUND, from: GameRoundState.NOT_STARTED, to: GameRoundState.WAITING_UPDATE },
-    { event: GameEvents.ROUND_STATE_FETCHED, from: GameRoundState.WAITING_UPDATE, to: GameRoundState.PLAYER_TURN_FROM_SERVER },
+    { event: GameEvents.START_ROUND, from: GameRoundState.NOT_STARTED, to: GameRoundState.WAITING_UPDATE, onEvent: setGameId },
+    { event: GameEvents.ROUND_STATE_FETCHED, from: GameRoundState.WAITING_UPDATE, to: GameRoundState.PLAYER_TURN_FROM_SERVER, onEvent: updateGameRound },
     { event: GameEvents.PLAYER_MOVED, from: GameRoundState.PLAYER_1_MOVES, to: GameRoundState.WAITING_UPDATE },
 
-    { event: GameEvents.START_ROUND, from: GameRoundState.PLAYER_1_WON, to: GameRoundState.WAITING_UPDATE },
-    { event: GameEvents.START_ROUND, from: GameRoundState.PLAYER_2_WON, to: GameRoundState.WAITING_UPDATE },
-    { event: GameEvents.START_ROUND, from: GameRoundState.DRAW, to: GameRoundState.WAITING_UPDATE },
+    // Will be available once issue with multiple timers is resolved
+    // { event: GameEvents.START_ROUND, from: GameRoundState.PLAYER_1_WON, to: GameRoundState.WAITING_UPDATE },
+    // { event: GameEvents.START_ROUND, from: GameRoundState.PLAYER_2_WON, to: GameRoundState.WAITING_UPDATE },
+    // { event: GameEvents.START_ROUND, from: GameRoundState.DRAW, to: GameRoundState.WAITING_UPDATE },
 ]
 
 const tictactoeFSM = function (state = initialState, action) {
```

Comment:
This function still feels complicated and non-obvious to me. 
Some of it has to do with the wording:
How come a variable named transition-to gets assigned from “mapNewState(...)”?
Which is then assigned back to gameState in the object?

Reply:
Simplified the approach, getting rid of transitions as list and not obvious methods to create new round state